### DISCUSSION TITLE
Total noob, thought I'd try it out

### DISCUSSION THREAD

**Discussion by BronsonCarder** [SebLague/Chess-Challenge#223] [General]
I took C# in college... someday I'd like to make video games, but right now, even this simple chess bot is pretty reasonably over my head. lol

But, I made some solid breakthroughs today. Not only am I beating Evilbot basically every time, but my v2 beats my v1 almost every time as well. lol

Frankly, I didn't think I would end up making anything worthwhile... I decided that I wasn't going to learn any of the established techniques... both because it's mostly way over my head anyway, but also because I felt that figuring out my own approach was more in the spirit of the challenge.
I've learned a LOT and I can't wait to see this challenge close out and see what everyone came up with. 
I'm also kind of running out of ideas at this point... Not sure if there will be a lot of upgrades from here, but, I'm honestly pretty proud of what I have.

**Comment by BronsonCarder** [comment]
About my approach:

I essentially just took the idea of piece value determining move value and ran with it. 
I'm making a list of the values of all legal movies, adjusting those values by a bunch of factors, and then outputting one of the moves that tie for first place at random. 
The part of this I'm most proud of is my DangerValue calculation. Basically it assigns a value to how "dangerous" the board state is, before and after the move, based on the value of the highest value piece that is at risk (as well as the number of pieces that are at risk).

Then, you subtract the After from the Before. 

So, if before there's nothing in danger, and after there's a queen in danger:
900 - 0 = 900. Subtract that from the value of the move, and it's basically just not gonna do it unless the move is otherwise extremely valuable. 
But, and this is kind of the awesome part, you kind of get a blocking incentivization system for free; if before there's a queen in danger, and after there's nothing in danger, then:
0 - 900 = -900. Subtract a negative and you add. 
So, with this one function you get disincentivizing risky moves, and incentivizing blocking valuable pieces. 
I bet there's a much fancier name for what I've come up with here. lmao

**Comment by SebLague** [comment]
That sounds great, I'm happy you've been enjoying the challenge!
